pymbar/tests/test_mbar.py

```diff
@@ -87,6 +87,19 @@ def test_mbar_free_energies():
         z = (fe - fe0) / fe_sigma
         eq(z / z_scale_factor, np.zeros(len(z)), decimal=0)
 
+        # now test the bootstrap uncertainty
+        mbar = MBAR(u_kn, N_k, nbootstraps = 40)
+        results = mbar.getFreeEnergyDifferences(uncertainty_method='bootstrap', return_dict=True)
+        fe = results['Delta_f']
+        fe_sigma = results['dDelta_f']
+        fe, fe_sigma = fe[0,1:], fe_sigma[0,1:]
+
+        fe0 = test.analytical_free_energies()
+        fe0 = fe0[1:] - fe0[0]
+
+        z = (fe - fe0) / fe_sigma
+        eq(z / z_scale_factor, np.zeros(len(z)), decimal=0)
```

Comment:
@mrshirts : I'm reviewing the test more closely now, and I'm wondering whether this is actually a useful test -- its mostly testing that the mbar solved free energies are close to the analytical free energies. However, the bootstrapped free energy matrix `self.f_k_boots` is never actually used to the free energy, its only used to update the uncertainty. Is there a simple way I add a test that checks the bootstrapped uncertainty is reasonable?

Reply:
Ah, yeah, these are a port of the old test.  I am still trying to figure out the best way to test the bootstraps for #448 - if you see that PR, since bootstrap error is different in that it needs to be initialized differently.

Reply:
Oh, I see these are already initialized - the newer pytests are much more extensive. and do things differently than the older tests.

It's doing SOMETHING, in that you calculate the number of standard deviations away the error is. But I'm not exactly what the last test is doing.  It's not a very good test - if it's a good error estimate, Z would be Gaussian normal distributed with standard deviation one, but that's not really testing that.   

My instinct is to focus on making the pymbar4 tests better, though.

Reply:
@mrshirts : Final thing (I think) for this PR -- what do you think of this test to check that the bootstrapping is working properly:

```python
from pymbar import MBAR
from pymbar.testsystems import HarmonicOscillatorsTestCase

# Generate harmonic oscillator with plenty of samples for each state
testcase = HarmonicOscillatorsTestCase()
[x_kn, u_kn, N_k, s_n] = testcase.sample(N_k=[10000, 10000, 10000, 10000, 10000])

# Compute non-bootstrapped uncertainty
mbar = MBAR(u_kn, N_k)
results = mbar.getFreeEnergyDifferences(compute_uncertainty=True, return_dict=True)
stderr = results['dDelta_f'][0, -1]

# Compute bootstrapped uncertainty
mbar_boots = MBAR(u_kn, N_k, nbootstraps=200, solver_tolerance=1e-6, initialize='BAR')
results_boots = mbar_boots.getFreeEnergyDifferences(compute_uncertainty=True, uncertainty_method='bootstrap', return_dict=True)
stderr_boots = results_boots['dDelta_f'][0, -1]

assert stderr_boots < 6 * stderr, f"Boostrapped standard error ({stderr_boots}) is more than six times the unbootstrapped standard error ({6* stderr})"

```